optuna/study/study.py

```diff
@@ -928,6 +946,37 @@ def _pop_waiting_trial_id(self) -> Optional[int]:
 
         return None
 
+    def _should_skip_enqueue(self, params: Dict[str, Any]) -> bool:
+
+        for trial in self.get_trials(deepcopy=False):
+            trial_params = trial.system_attrs.get("fixed_params", trial.params)
+            if trial_params.keys() != params.keys():
+                # Can't have repeated trials if different params are suggested.
+                continue
+
+            repeated_params: List[bool] = []
+            for param_name, param_value in params.items():
+                existing_param = trial_params[param_name]
+                if not isinstance(param_value, type(existing_param)):
+                    # Enqueued param has distribution that does not match existing param
+                    # (e.g. trying to enqueue categorical to float param).
+                    # We are not doing anything about it here, since sanitization should
+                    # be handled regardless if `skip_if_exists` is `True`.
+                    continue
```

Comment:
[nits] If you add `repeated_params.append(False)` here, you do not need to check `len(repeated_params) > 0`. And you can just return `all(repeated_params)` at the end.

Reply:
Thanks for the suggestion. I've removed length check in 3b99b8436bcf2e95836369b1b2c4d96d28c55b13, but otherwise early breakout logic remains unchanged, otherwise we would always return after checking only first trial.